test/client-getAll.test.ts

```diff
@@ -124,3 +126,68 @@ test("merges params and default params if provided", async (t) => {
 	t.deepEqual(res, allDocs);
 	t.is(res.length, 3 * 3);
 });
+
+test("throttles requests past first page", async (t) => {
+	const numPages = 3;
+	const repositoryResponse = createRepositoryResponse();
+	const pagedResponses = createQueryResponsePages({
+		numPages,
+		numDocsPerPage: 3,
+	});
+
+	server.use(
+		createMockRepositoryHandler(t, repositoryResponse),
+		createMockQueryHandler(t, pagedResponses, undefined, {
+			ref: getMasterRef(repositoryResponse),
+			pageSize: 100,
+		}),
+	);
+
+	const client = createTestClient(t);
+
+	const startTime = Date.now();
+	await client.getAll();
+	const endTime = Date.now();
+
+	const totalTime = endTime - startTime;
+	const maxTime = numPages * GET_ALL_THROTTLE_THRESHOLD;
+	const minTime = maxTime - GET_ALL_THROTTLE_THRESHOLD;
+
+	// The total time should be between (# of pages - 1) and # of pages
+	// multiplied by the throttle threshold duration. This effectively checks
+	// that each request after page 1 is delayed by at least the threshold
+	// amount.
```

Comment:
True, in the test I assume the request will be resolved quickly. Otherwise it may be difficult to test this without a deterministic delay.

MSW has a `delay` function that could make this more explicit: https://mswjs.io/docs/api/context/delay

Reply:
The mock request handlers now use `delay` so we can more deterministically track how long a request will take to resolve.

We still need to account for the time it takes for the JS to run, so `NETWORK_REQUEST_DURATION_TOLERANCE` is defined at the top of the test file. Basically `NETWORK_REQUEST_DURATION_TOLERANCE` is the amount of time it takes for the method to complete its work between network requests, which will be machine-specific.

Reply:
Great job! Yeah I was nitpicking sorry ^^'